src/utils/chekedFetch.js

```diff
@@ -5,17 +5,22 @@ class HttpError extends Error {
     }
 }
 
-const checkResponseStatusOrThrowError = (response, errorMessage) => {
-    const code = response.status
-    if(code < 200 || code > 299) {
-        throw new HttpError(code, errorMessage || "Something went wrong")
+const checkResponseStatusOrThrowError = (result, errorMessage) => {
+    const {status, data} = result
+    console.log(status, data)
+    if(status < 200 || status > 299) {
+        throw new HttpError(status, data.message || errorMessage || "Something went wrong")
     }
 }
 
 export const checkedFetch = (url, data, errorMessage) => {
     return fetch(url, data)
         .then(response => {
-            checkResponseStatusOrThrowError(response, errorMessage)
-            return response
+            const responsev2 = response.clone()
+            return responsev2.json().then(data => ({status: response.status, data: data, response: response}))
+        })
```

Comment:
Co to robi??

Reply:
Ogólnie żeby teoretycznie lepiej wyświetlać błędy (czyli message z backu) musimy mieć dostęp do danych. Ten `then` rozbija response na statusCode, dane i response. Niestety funkcja niżej wymaga zwrócenia response'a, a response.json() możesz wykonać tylko raz, więc musiałem zklonować oryginalny response, żeby rozłożyć na dane klona, a oryginalny reponse zachować do zwrócenia.

Reply:
Mogę tylko trochę się czepnąć, że "responsev2" nie jest najszczęśliwsza nazwa XD

Reply:
Ona jest akurat tak głęboko, że już jej nigdy najprawdopodobniej nie zobaczysz, ale mogę zmienić XD